frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts

```diff
@@ -451,7 +458,7 @@ export type OnFilterChange = (type: string, value: FilterValue) => void;
 export type ListPageFilterProps<D = any> = {
   data: D;
   loaded: boolean;
-  rowFilters?: RowFilter[];
+  rowFilters?: (RowFilter<D> | RowSearchFilter<D>)[];
```

Comment:
This change shouldn't be necessary.

The `RowFilter` type is a union of all specific filter types. Code that interprets this prop differentiates between specific filters by detecting specific (non-optional) properties:
- `isMatch` for `RowMatchFilter`
- `reducer` for `RowReducerFilter`

The proposed `RowSearchFilter` should be differentiated via some of its required properties. This is problematic because the `placeholder?: string` is marked as optional.

Given the above, I'd recommend leaving the existing `rowFilters` prop as-is (also leaving the existing `RowFilter` union type as-is) and introduce a new optional prop here:

```ts
rowSearchFilters?: RowSearchFilter<D>[]
```

Reply:
Yes, this is right. We don't need `RowFilter | RowSearchFilter`. The main difference is that `RowSearchFilter` does not have `items` as in the search there aren't options to choose from.